src/main/java/com/auctionapp/api/service/UserService.java

```diff
@@ -25,30 +36,72 @@ public UserDto getUser(final String email) {
         throw new RuntimeException("User with email " + email + " does not exist!");
 	}
 
+	public UserDto getUserById(final UUID id) {
+		Optional<User> user = userRepository.findById(id);
+        if (user.isPresent()) {
+			return toPayload(user.get());
+		}
+        throw new RuntimeException("User with id " + id + " does not exist!");
+	}
+
+	public void deactivateUser(final UUID id) {
+		Optional<User> user = userRepository.findById(id);
+		user.get().setStatus(Status.INACTIVE);
+		userRepository.save(user.get());
```

Comment:
What if there is no user with such id? This will throw an exception. If that is the behavior you want, make a `get` instead of a `find` method. Generally, we use `get` when we are sure a resource exists or should exist and we want to throw an exception if it doesn't, and find when whether the resource exists or not is something we're not sure about.